packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -0,0 +1,296 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import 'button.dart';
+
+// Padding around the line at the edge of the text selection that has 0 width and
+// the height of the text font.
+const double _kHandlesPadding = 18.0;
+// Minimal padding from all edges of the selection toolbar to all edges of the
+// viewport.
+const double _kToolbarScreenPadding = 8.0;
+const double _kToolbarHeight = 36.0;
+
+const Color _kToolbarBackgroundColor = const Color(0xFF2E2E2E);
+const Color _kToolbarDividerColor = const Color(0xFFB9B9B9);
+const Color _kHandlesColor = const Color(0xFF146DDE);
+
+const EdgeInsets _kToolbarButtonPadding = const EdgeInsets.symmetric(vertical: 10.0, horizontal: 21.0);
+
+const Size _kToolbarTriangleSize = const Size(18.0, 9.0);
+const double _kToolbarBorderRadius = 7.5;
+
+const TextStyle _kToolbarButtonFontStyle = const TextStyle(
+  fontSize: 14.0,
+  letterSpacing: -0.11,
+  fontWeight: FontWeight.w300,
+);
+
+const Widget _kOnePhysicalPixelVerticalDivider = const DecoratedBox(
+  decoration: const BoxDecoration(color: _kToolbarDividerColor),
+  child: const SizedBox(width: 0.5),
```

Comment:
Double checked. It is indeed exactly one physical pixel on all devices. Changed from constant to read the device pixel ratio.